### PR TITLE
Fix SHADOWS_DISABLED flag in GLES2

### LINKED ISSUE #27061: GLES2 "Do not Receive Shadows" is not working
**Godot version:**
3.1 stable

**OS/device including version:**
Windows 10

**Issue description:**
The spatial material has an option "Do not Receive Shadows". Shadows are rendered onto this mesh even if this option is checked. It's working as intended on GLES3.

GLES2:
gles2

GLES3:
gles3

**Steps to reproduce:**
- Create a new 3D scene 
- add a light source
- add two meshes, one which is casting a shadow and one which is receiving the shadow
- check "Do not Receive Shadows" in the options of the spatial material of the receiving mesh

**Minimal reproduction project:**

[ReceiveShadowBug.zip](https://github.com/godotengine/godot/files/2968193/ReceiveShadowBug.zip)

### PR DESCRIPTION
**Pull request by SouzaGuilherme**
Closes #27061

### CODE REVIEW

**Comment by clayjohn** [comment]
You will need to add this in a few other places as well. Just search through the file for ``USE_SHADOW``. The shadow code for the different types of lights are all in different sections. So you need to make sure shadows get disabled for all of them. Currently, this is just disabling shadows for directional lights, shadows for omni lights and spotlights still need to be disabled.

**Comment by qarmin** [comment]
Shouldn't you use 
```
#ifndef SHADOWS_DISABLED
```
instead
```
#if !defined(SHADOWS_DISABLED)
```
?

**Comment by clayjohn** [comment]
@qarmin Usually it is ``ifndef`` when there is one item and then ``!defined()`` when there is mulitple, however GLES3 uses ``!defined(SHADOWS_DISABLED)`` everytime.

https://github.com/godotengine/godot/blob/df18c8c413306870871a1bf257182f2cd65016f3/drivers/gles3/shaders/scene.glsl#L1261

https://github.com/godotengine/godot/blob/df18c8c413306870871a1bf257182f2cd65016f3/drivers/gles3/shaders/scene.glsl#L1323

https://github.com/godotengine/godot/blob/df18c8c413306870871a1bf257182f2cd65016f3/drivers/gles3/shaders/scene.glsl#L1822

In this case it is likely best to combine it with the ``USE_SHADOW`` define. as ``#if defined(USE_SHADOW) && !defined(SHADOWS_DISABLED)``

File: drivers/gles2/shaders/scene.glsl [hunk 1]
```diff
@@ -1466,9 +1466,6 @@ void main() {
 	float anisotropy = 0.0;
 	vec2 anisotropy_flow = vec2(1.0, 0.0);
 	float sss_strength = 0.0; //unused
-	// gl_FragDepth is not available in GLES2, so writing to DEPTH is not converted to gl_FragDepth by Godot compiler resulting in a
-	// compile error because DEPTH is not a variable.
-	float m_DEPTH = 0.0;
```
Comment (clayjohn) [inline drivers/gles2/shaders/scene.glsl]: You shouldn't mix another change into a PR. Always stick to one topic, if you want to make other changes you should make multiple PRs.

In this case, removing ``m_DEPTH`` will cause a shader compile error when users try to access ``DEPTH`` in the fragment shader.

Comment (SouzaGuilherme) [reply drivers/gles2/shaders/scene.glsl]: It was my mistake. I fixed and also squashed the commits.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**